scripts/spark/added/launch.sh

```diff
@@ -9,6 +9,22 @@ function check_reverse_proxy {
     fi
 }
 
+function handle_term {
+    echo Received a termination signal
+    # If we've saved a PID for a subprocess, kill that first before
+    # trying to delete the cluster
+    if [ -n "$PID" ]; then
+        echo "Stopping subprocess $PID"
+        kill -TERM $PID
+        wait $PID
+        echo "Subprocess stopped"
+    fi
+    exit 0
+}
+
+
+trap handle_term TERM INT
```

Comment:
Standard shutdown from kube is TERM followed by KILL after 30 seconds if the pod is still there.  INT will get us ctrl-c if we're playing with images directly so we've got the usual cases covered.  What other signals do you think we should handle?